internal/bminventory/inventory.go

```diff
@@ -2987,6 +2987,31 @@ func (b *bareMetalInventory) processDiskSpeedCheckResponse(ctx context.Context,
 	return b.hostApi.SetDiskSpeed(ctx, h, diskPerfCheckResponse.Path, diskPerfCheckResponse.IoSyncDuration, exitCode, nil)
 }
 
+func (b *bareMetalInventory) updateDomainNameResolutionResponse(ctx context.Context, host *models.Host, domainResolutionResponseReporter string) error {
+	var (
+		err                      error
+		domainResolutionResponse models.DomainResolutionResponse
+	)
+	log := logutil.FromContext(ctx, b.log)
+	log.Debugf("The address for domain name resolotion on host %s are: %s", host.ID.String(), domainResolutionResponse)
+
+	if err = json.Unmarshal([]byte(domainResolutionResponseReporter), &domainResolutionResponse); err != nil {
+		log.WithError(err).Warnf("Json unmarshal domain name resolution of host %s", host.ID.String())
+		return err
+	}
+	if len(domainResolutionResponse.Resolutions) == 0 {
+		err = errors.Errorf("Domain name resolution of host %s is empty", host.ID.String())
+		log.WithError(err).Warn("Check host DNS configuration")
+		return err
+	}
+	if err = b.db.Model(&common.Host{}).Where("id = ? and cluster_id = ?", host.ID.String(),
+		host.ClusterID.String()).UpdateColumn("domain_name_resolution", domainResolutionResponseReporter).Error; err != nil {
```

Comment:
We should check if the resolution is not the same as before.  If it isn't then we should issue Update (not UpdateColumn), because we want the UpdatedAt field to be updated in that case

Reply:
Wouldn't it make sense to always bump `UpdatedAt`? To clarify that the DNS resolution information is up-to-date

Reply:
We need to avoid making updates that are not required. This helps reduce the amount of hosts/clusters that the monitors need to handle.
This was an important part of the performance improvement that @ori-amizur made